doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+
+#### Result value
+
+The result is `.true.` if the bit at `pos`, in `self` is set,
+otherwise it is `.false.`. If `pos` is outside the range
+`0... bits(self)-1` the result is `.false.`.
+
+#### Example
+
+```fortran
+    program demo_test
+        use stdlib_bitsets
+        type(bitset_large) :: set0
+        call set0 % init(166)
+        call set0 % not()
+        if ( set0 % all() ) write(*,*) 'SET0 is properly initialized.'
+        call set0 % clear(165)
+        if ( .not. set0 % test(165) ) write(*,*) 'Bit 165 is cleared.'
+        call set0 % set(165)
+        if ( set0 % test(165) ) write(*,*) 'Bit 165 is set.'
+    end program demo_test
+```
+
+### `to_string` - represent a bitset as a binary literal
+
+### Status
+
+Experimental
+
+#### Description
+
+Represents the value of `self` as a binary literal in `string`.
+
+#### Syntax
+
+`call self % [[bitset_type(class):to_string(bound)]](string[, status])
+
+#### Class
+
+Subroutine
+
+#### Arguments
+
+`self`: shall be a scalar expression of class `bitset_type`. It is an
+`intent(in)` argument.
+
+`string`: shall be a scalar default character variable of allocatable
+length. It is an `intent(out)` argument. On return it shall hav a
+*binary-literal* representation of the bitset `self`.
+
+`status` (optional): shall be a scalar default integer variable. It is
+an `intent(out)` argument. If present it shall have either the value
+`success` or `alloc_fault`. If absent and it would have had the value
+`alloc_fault` then processing will stop with an informative test as
+the stop code. The values have the following meanings:
+
+`success` - no problem found.
+
+`alloc_fault` - allocation of `string` failed.
+
+
+#### Example
+
+```fortran
+    program demo_to_string
+        use stdlib_bitsets
+        character(*), parameter :: &
+            bits_all = '111111111111111111111111111111111'
+        type(bitset_64) :: set0
+        character(:), allocatable :: new_string
+        call set0 % init(33)
+        call set0 % not()
+        call set0 % to_string( new_string )
+        if ( new_string == bits_all ) then
+            write(*,*) "TO_STRING transferred BITS0 properly" // &
+                " into NEW_STRING."
+        end if
+    end program demo_to_string
+```
+
+### `value` - determine the value of a bit
+
+#### Status
+
+Experimeental
+
+#### Description
+
+Determines the value of the bit at position, `pos`, in `self`.
+
+#### Syntax
+
+`result = self % [[bitset_type(class):value(bound)]](pos)`
+
+#### Class
+
+Elemental function.
+
+#### Arguments
+
+`self`: shall be a scalar expression of class `bitset_type`. It is an
+`intent(in)` argument.
+
+`pos`: shall be a scalar integer expression of kind `bits_kind`. It is
+an `intent(in)` argument.
+
+#### Result character
+
+The result is a default logical scalar.
+
+#### Result value
+
+The result is one if the bit at `pos` in `self` is set, otherwise it
+is zero. If `pos` is outside the range `0... bits(set)-1` the result
+is zero.
+
```

Comment:
```suggestion
#### Result value

The result is a default logical scalar.
The result is 1 if the bit at `pos` in `self` is set, otherwise it
is 0. If `pos` is outside the range `0... bits(set)-1` the result
is 0.

```

Again, does it make sense to return 0 if pos is outside the range?